supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalJobExecutionFSM.scala

```diff
@@ -0,0 +1,248 @@
+package cromwell.backend.impl.local
+
+import java.nio.file.{FileSystems, Path, Paths}
+
+import akka.actor.FSM.NullFunction
+import akka.actor.{PoisonPill, Props, LoggingFSM}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptor
+import cromwell.backend.BackendJobExecutionActor.{BackendJobExecutionAbortedResponse, BackendJobExecutionFailedResponse, BackendJobExecutionResponse, BackendJobExecutionSucceededResponse}
+import cromwell.backend.BackendLifecycleActor.{BackendJobExecutionAbortFailedResponse, BackendJobExecutionAbortSucceededResponse, JobAbortResponse}
+import cromwell.backend.impl.local.LocalJobExecutionFSM._
+import cromwell.core.CallContext
+import org.slf4j.LoggerFactory
+import wdl4s._
+import wdl4s.values.{WdlFile, WdlValue}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, Promise}
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.{Failure, Success, Try}
+
+object LocalJobExecutionFSM {
```

Comment:
I asked for this FSM... but it looks like you can't extend the JobExecutionActor base trait with FSM. So the parent's job it interfacing with the engine. This one's job is actually doing the work as an FSM.

If it turns out in the future that every JobExecutionActor wants to be an FSM, maybe we can modify the trait and remove the need for the thin layer